### DISCUSSION TITLE
Fill the whole container without resize

### DISCUSSION THREAD

**Discussion by vdemcak** [uber/react-map-gl#1882] [Q&A]
Hey,  I've came across an issue. For some reason, most of the time the map doesn't fill the whole viewport, instead it covers only a tiny bit of it, but when i resize the window it fixes itself. 

Currently, the map is placed in a view grid that covers the whole screen. I've already tried calling map.resize() on load, but I don't like the fact that you can still see the wrongly sized map before it fully loads in and resizes itself.

Did anyone have issues with this and did you manage to fix it?

NOTE: For some reason, the map works perfectly fine in dev, it is only broken in production. I'm using React.js + Vite, for package versions I'm on latest versions for both mapbox-gl and react-map-gl.

**Comment by Pessimistress** [comment]
mapbox-gl does not automatically resize even with the `trackResize` option set to `true`. It only knows that the container size has changed on the window's `resize` event. So if you have a container whose size is updated after the initial mount, then the map will show the wrong size until you call `map.resize()`.

My guess is that you are programmatically setting the container size, or have some sort of CSS transition/animation when your app initially loads.
